### PR TITLE
fix: headless mode

### LINKED ISSUE #53: chore: automatic tests when a PR is opened
To help with the flow of contributions to the project, it would be nice to have the tests run when contributions are submitted. Especially since it usually helps us test against other filesystems/architectures where there might be hiccups with certain implementation approaches.

This have been attempted once and it ran into an issue where the popup's weren't being detected. 
https://github.com/TenKeyLabs/dappwright/blob/e3aecc61853fb652a590842b4feda51f58f8a08a/.github/workflows/test.yaml#L27-L51
Output - https://github.com/TenKeyLabs/dappwright/actions/runs/3946033767/jobs/6753437903

**Additional context**
Have had no luck replicating the failures via Docker using the Ubuntu image and following the test steps.

### PR DESCRIPTION
**Pull request by osis**
The current headless mode was receiving an error along the lines of `net::ERR_ABORTED chrome-extension://`

This fixes headless mode by adjusting to the current state of is chrome's headless flags.

TLDR on chrome headless - There is `--headless` but that doesn't allow you to load extensions. So they created `--headless=chrome` which then migrated to `--headless=new` which is the flag that we're looking for.

This also changes the default behaviour of the headless flag to match the playwright defaults, which is to default to headless mode unless explicitly told otherwise. Which means this works out of the box with the `playwright test --headless` flag.

Also adds the wallet tests to the github action. The rest of the test will be added when the example application had been extracted since it's having some issues executing in the actions env.

### PR Checklist

- [x] I have run linter locally
- [x] I have run unit and integration tests locally
- [x] Update configuration the newest version (readme and const)
- [x] Rebased to master branch / merged master

Fixes #53

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -24,3 +24,31 @@ jobs:
...
+
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile
+
+      - name: Test
+        env:
+          GITHUB_TOKEN: ${{ github.token }}
+        run: |
+          npx playwright install --with-deps chromium
+          xvfb-run --auto-servernum yarn run test:ci
```
Comment (bobby-chan) [inline .github/workflows/test.yaml:48]: Do we need to do `xvfb-run` anymore?

File: package.json [hunk 2]
```diff
@@ -26,6 +26,7 @@
     "lint": "eslint --color --ext .ts src/ test/",
     "lint:fix": "yarn run lint --fix",
     "test": "playwright test test/",
+    "test:ci": "playwright test --headed --timeout 180000 test/",
```
Comment (bobby-chan) [inline package.json:29]: Probably don't need `--headed`

File: test/dapp/server.ts [hunk 3]
```diff
@@ -32,9 +32,9 @@ export async function stop(): Promise<void> {
   await chainNode.close();
 }
 
-async function waitForGanache(): Promise<Provider> {
+export async function waitForGanache(): Promise<Provider> {
   console.log('Starting ganache...');
-  chainNode = ganache.server({ chain: { chainId: 31337 }, wallet: { seed: 'asd123' }, logging: { quiet: true } });
+  chainNode = ganache.server({ chain: { chainId: 31337 }, wallet: { seed: 'asd123' }, logging: { quiet: false } });
```
Comment (bobby-chan) [inline test/dapp/server.ts:37]: Can probably turn `quiet` back to true

**Review by bobby-chan — APPROVED** [review]
Not sure why the yarn.lock file has so much stuff in it, but otherwise seems fine

File: .github/workflows/test.yaml (hunk 1 above)
Comment (osis) [reply .github/workflows/test.yaml:48]: Unfortunately for now. The headed mode seems to bring a higher degree of uncertainty to the tests in github actions. Stable on localhost though. Needs more investigation.
